integration-tests/tests/utils/clp_mode_utils.py

```diff
@@ -0,0 +1,97 @@
+"""Provides utilities related to the user-level configurations of CLP's operating modes."""
+
+from collections.abc import Callable
+from typing import Any
+
+from clp_py_utils.clp_config import (
+    CLPConfig,
+    COMPRESSION_SCHEDULER_COMPONENT_NAME,
+    COMPRESSION_WORKER_COMPONENT_NAME,
+    DB_COMPONENT_NAME,
+    GARBAGE_COLLECTOR_COMPONENT_NAME,
+    Package,
+    QUERY_SCHEDULER_COMPONENT_NAME,
+    QUERY_WORKER_COMPONENT_NAME,
+    QueryEngine,
+    QUEUE_COMPONENT_NAME,
+    REDIS_COMPONENT_NAME,
+    REDUCER_COMPONENT_NAME,
+    RESULTS_CACHE_COMPONENT_NAME,
+    StorageEngine,
+    WEBUI_COMPONENT_NAME,
+)
+
+
+def _to_container_basename(name: str) -> str:
```

Comment:
```suggestion
def _to_docker_compose_service_name(name: str) -> str:
```
Do you agree this would be better? Also add a note saying this would be replaced by a formalized mapping between component <-> service